app/views/portal/show.rb

```diff
@@ -783,154 +753,173 @@ def iiif_manifesto(identifier, collection)
-        end
+      item = {
+        media_type: media_type,
+        rights: simple_rights_label_data(media_rights),
+        downloadable: item_downloadable?(web_resource_url, mime_type, media_type, media_rights),
+        playable: item_playable?(web_resource_url, mime_type, media_type),
+        thumbnail: item_thumbnail(media_type),
+        play_url: manifesto.present? ? manifesto : web_resource_url
+      }
 
-        media_type = media_type(web_resource_url)
-        media_type = media_type || render_document_show_field_value(document, 'type')
-        media_type = media_type.downcase
+      player = item_player(media_type, mime_type)
+      if player.nil?
+        item[:is_unknown_type] = render_document_show_field_value(web_resource, 'about')
+      else
+        item[:"is_#{player}"] = true
+      end
 
-        if media_type == 'audio' && mime_type == 'application/octet-stream'
-          if !web_resource_url.index('.mp3').nil?
-            mime_type = 'audio/mpeg'
-          end
-        end
+      item[:download] = {
+        url: media_proxy_download_url(web_resource_url, mime_type),
+        text: t('site.object.actions.download')
+      }
+      item[:technical_metadata] = item_technical_metadata(web_resource, mime_type)
 
-        identifier = render_document_show_field_value(document, 'proxies.dcIdentifier')
-        collection = render_document_show_field_value(document, 'europeanaCollectionName')
-        manifesto = iiif_manifesto(identifier, collection)
+      if web_resource_url == edm_resource_url
+        item[:thumbnail] = edm_preview
+        item[:is_current] = true
+      end
 
-        if manifesto
-          media_type = 'iiif'
-        end
+      item
+    end
 
-        item = {
-          media_type: media_type,
-          rights: simple_rights_label_data(media_rights),
-          downloadable: true,
-          playable: edm_is_shown_by_download_url.present?,
-          thumbnail: edm_preview
-        }
+    def edm_resource_url
+      @edm_resource_url ||= render_document_show_field_value(document, 'aggregations.edmIsShownBy')
+    end
 
-        if download_disabled(media_rights)
-          item[:downloadable] = false
-        end
+    def edm_preview
+      @edm_preview ||= render_document_show_field_value(document, 'europeanaAggregation.edmPreview', tag: false)
+    end
 
-        # disable play / download
+    def media_items
+      items = salient_web_resources.map do |web_resource|
+        web_resource_media_item(web_resource)
+      end
 
-        if mime_type == 'video/mpeg'
-          item[:playable] = false
-        end
+      {
+        required_players: item_players(items),
+        single_item: items.size == 1,
+        empty_item: items.size == 0,
+        items: items,
+        more_thumbs_url: request.original_url.split('.html')[0] + '/thumbnails.json'
+      }
+    end
 
-        if mime_type == 'audio/flac'
-          item[:playable] = true
-        end
+    def salient_web_resources
```

Comment:
Assignment Branch Condition size for salient_web_resources is too high. [22.45/15]